cypress-io/cypress
# chore: remove check-ts from Gulp

- Closes https://github.com/cypress-io/cypress/issues/27491

### Additional details


This PR removes the Gulp task for `check-ts` and runs it using Lerna. It also adds the step as a cacheable operation, which means that when we run `yarn check-ts`, the step will only run for packages that have had source changes since the task was last run.

It also runs all `check-ts` scripts in parallel which reduces the time that it takes to run these checks.

### Steps to test


- Check [the CI job](https://app.circleci.com/pipelines/github/cypress-io/cypress/55274/workflows/326180bc-ac13-4bee-87ce-5c6f07ec8198/jobs/2287075) for this branch (notice the 4m26s runtime compared to the p95 8m30s for this job on `develop` 🚀 )
- Run `yarn check-ts` locally, make some changes to create some TS errors and verify that running this step catches them

### How has the user experience changed?


n/a

### PR Tasks


- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
